core/integration/services_test.go

```diff
@@ -109,6 +108,23 @@ func TestContainerHostnameEndpoint(t *testing.T) {
 		require.Equal(t, hn+":8000", ep)
 	})
 
+	t.Run("hostname and endpoint force default command", func(t *testing.T) {
+		srv := c.Container().
+			From("python").
+			WithExposedPort(8000).
+			WithDefaultArgs(dagger.ContainerWithDefaultArgsOpts{
+				Args: []string{"python", "-m", "http.server"},
+			})
+
+		hn, err := srv.Hostname(ctx)
+		require.NoError(t, err)
+
+		ep, err := srv.Endpoint(ctx)
+		require.NoError(t, err)
+
+		require.Equal(t, hn+":8000", ep)
```

Comment:
This seems like it would pass before the change, too. Maybe we could assert that the hostname is `==` to `srv.WithExec(nil).Hostname(ctx)` just to be super explicit?

edit: oh, I see you do that in another test below. Maybe that assertion could be moved here?

Reply:
> This seems like it would pass before the change, too.

Not really. Before it would error. And this is making sure that both are adding `WithExec(nil)`, not just one.

Notice that this test is replicating the one just above it, just replacing `WithExec` with `WithDefaultArgs`.

> Maybe that assertion could be moved here?

Yes, if that makes it clearer. I added that later on. 🙂